### DISCUSSION TITLE
Ambiguous use of user-defined command:

### DISCUSSION THREAD

**Discussion by Hugo-Persson** [CopilotC-Nvim/CopilotChat.nvim#40] [Q&A] [answered]
I get the following error when i try to run CopilotChatExplain, do anyone have any suggestion on how to fix?

Full error:
```
Error executing Lua callback: vim/_editor.lua:0: nvim_exec2(): Vim:E464: Ambiguous use of user-defined command: CopilotChat Explain how it works.
stack traceback:
	[C]: in function 'nvim_exec2'
	vim/_editor.lua: in function 'cmd'
	...hare/nvim/lazy/CopilotChat.nvim/lua/CopilotChat/init.lua:32: in function <...hare/nvim/lazy/CopilotChat.nvim/lua/CopilotChat/init.lua:31>
	```

**Comment by GitMurf** [comment, marked as answer]
@Hugo-Persson see here: https://github.com/CopilotC-Nvim/CopilotChat.nvim/issues/21#issuecomment-1927329576

**Comment by Hugo-Persson** [comment]
Thank you, @GitMurf, stupid questions, but I do not have the editor command :UpdateRemotePlugins, do I need to install some dependency?

**Reply by jellydn** [reply]
Hi @Hugo-Persson You could check this document from Neovim https://neovim.io/doc/user/remote_plugin.html#%3AUpdateRemotePlugins

**Reply by Hugo-Persson** [reply]
I read that but I do not understand what I need to install to get the command :UpdateRemotePlugins

**Reply by GitMurf** [reply]
@Hugo-Persson are you using something like LazyVim, astro, chad etc.? if so they may by default disable remote plugins and therefore that could be why you do not see it. See this here: https://github.com/jellydn/CopilotChat.nvim/issues/25#issuecomment-1925194195

**Reply by Hugo-Persson** [reply]
Ok that fixed it for me, I used NvChad that disabled rplugin. Now I get the following error:
```
Unknown function: provider#python3#Prog
```
When running UpdateRemotePlugins

**Comment by GitMurf** [comment]
@jellydn I know there was a lot of discussion / gh issues in your previous repo on the ambiguous use of user defined command. Is this discussion here the best place to continue those conversations? I have tried absolutely everything and for sure have python working fine and integrated with neovim as my health checks all pass and I can run :!python fine from within neovim. Also the health check for CopilotChat passes as well.

It is just that when I do the UpdateRemotePlugins this is all I get (screenshot 1 below) and then nothing happens when I restart neovim and try the plugin. I get the ambiguous use of user defined command. To me it looks as if there are actually no remote plugins registered. Can you share what the `rplugin.vim` should look like? Mine seems to just be "empty" listing perl, node, ruby and python3 sections but nothing under the python3 section (see 2nd screenshot below). I assume this should be showing references to the CopilotChat remote plugins?

I am on Windows. Are there any confirmed successful uses of this plugin with anyone on Windows? Thanks for the great plugin and the support here!

![image](https://github.com/CopilotC-Nvim/CopilotChat.nvim/assets/64155612/1a61da02-84be-4efe-b8a5-886047fadbce)

![image](https://github.com/CopilotC-Nvim/CopilotChat.nvim/assets/64155612/a5e03154-ebc0-42be-863a-4c5971be5f13)

### Additionally here is what my health checks look like for Python. Everything seems to be configured properly:

![image](https://github.com/CopilotC-Nvim/CopilotChat.nvim/assets/64155612/1e1af910-0eb3-41df-9896-dd80aa520b50)

![image](https://github.com/CopilotC-Nvim/CopilotChat.nvim/assets/64155612/452c0cee-766d-435d-af60-4c20a00177ce)

![image](https://github.com/CopilotC-Nvim/CopilotChat.nvim/assets/64155612/19822dac-3f2f-44b3-b0e2-ca6ec9da6a78)

**Reply by jellydn** [reply]
> Are there any confirmed successful uses of this plugin with anyone on Windows? 

Yeah, there is a Windows user confirmed about it.
https://discord.com/channels/1200633211236122665/1200633212041449606/1203895226553339944

**Reply by GitMurf** [reply]
> > Are there any confirmed successful uses of this plugin with anyone on Windows? 
> 
> 
> 
> Yeah, there is a Windows user confirmed about it.
> 
> https://discord.com/channels/1200633211236122665/1200633212041449606/1203895226553339944
> 
> 
> 
> 

Thanks! I will see if maybe they have some advice :)

**Comment by GitMurf** [comment]
Ok I FINALLY figured it out! It definitely has to do with path issues on Windows compared to mac/linux so hopefully this will also help the CopilotChat maintainer team to make some changes in the codebase to better help with this process for Windows users :) cc @jellydn 

The details of the solution are here: https://github.com/CopilotC-Nvim/CopilotChat.nvim/discussions/27#discussioncomment-8438000

**Reply by GitMurf** [reply]
@jellydn can you update the "selected answer" to be this one. thanks!
